lib/dal/src/attribute/value.rs

```diff
@@ -2125,4 +2128,108 @@ impl AttributeValue {
 
         Ok(input_socket_ids)
     }
+    #[instrument(level = "info", skip_all)]
+    /// Get the morale equivalent of the prop path for a given attribute value id.
+    /// This includes the key/index in the path, unlike the path for a Prop which doesn't
+    /// include the key/index
+    pub async fn get_path_for_id(
+        ctx: &DalContext,
+        attribute_value_id: AttributeValueId,
+    ) -> AttributeValueResult<Option<String>> {
+        let mut parts = VecDeque::new();
+        let mut work_queue = VecDeque::from([attribute_value_id]);
+
+        while let Some(mut attribute_value_id) = work_queue.pop_front() {
+            let prop_name = match AttributeValue::prop_id_for_id(ctx, attribute_value_id).await {
+                Ok(prop) => Prop::get_by_id(ctx, prop).await?.name,
+                Err(_) => {
+                    // This means it's an input or output socket, which has no path and should just be empty string
+                    continue;
+                }
+            };
+
+            // get the parent to see if it's a Map or Array so we can add the necessary key/index to the prop path
+            if let Some(parent_attribute_value_id) =
+                AttributeValue::parent_attribute_value_id(ctx, attribute_value_id).await?
+            {
+                let parent_prop_id =
+                    AttributeValue::prop_id_for_id(ctx, parent_attribute_value_id).await?;
+                match Some(Prop::get_by_id(ctx, parent_prop_id).await?.kind) {
+                    Some(PropKind::Array) => {
+                        let order_of_thing_node = ctx
+                            .workspace_snapshot()?
+                            .incoming_sources_for_edge_weight_kind(
+                                attribute_value_id,
+                                EdgeWeightKindDiscriminants::Ordinal,
+                            )
+                            .await?
+                            .pop()
+                            .ok_or(AttributeValueError::NoOrderingNodeForAttributeValue(
+                                attribute_value_id,
+                            ))?;
+
+                        let order_node_weight = ctx
+                            .workspace_snapshot()?
+                            .get_node_weight(order_of_thing_node)
+                            .await?
+                            .get_ordering_node_weight()?;
+
+                        let order = order_node_weight.order();
```

Comment:
Same here as above.... Maybe on `WorkspaceSnapshot` itself though since "ordering nodes" do not belong to SI-specific entities.